## Liams Lab

Introducing: Python!!!

In [ ]:
for i in range(4):
    print(i)

hibhjhjhjhj

In [ ]:
for i in range(5): 
    print('Welcome, student {}  {}'.format(i, 'poop'))

### Data-types

- List
- Dictionary
- Set
- Tuple
- String

In [ ]:
#list
l = [1, 1, 2, 2, 3, 4, 5]
print('moo-ta-ble and freeeee {}'.format(l))

In [ ]:
#Don't mind me...
import time

#dictionary
d = {
        'key': 'value',
        'key2': 'value2'
    }
print('where my keys ...')
time.sleep(1)
print('looks left')
time.sleep(2)
print('looks right')
time.sleep(1)
print('OH! ',d.keys())
time.sleep(2)

print('but mah values...')
time.sleep(10000000)
print('stop me from looking for my values :(')

In [ ]:
#set
s = set(l)
s

In [ ]:
#tuple
t = (1, 2)
type(t)

In [ ]:
l

In [ ]:
h

In [ ]:
l

In [ ]:
#string
h = 'hello'
h

#### Write a program that prints out every number 1 - 10

In [ ]:
for i in range(10):
    print(i +1)

Count all the beans in a jar

In [ ]:
the_jar = ['bean', 'bean', 'bean', 'bean']

In [ ]:
counter=0
for bean in the_jar:
    
    counter=counter +1
counter

In [ ]:
for letter in 'Hellosaldkfjasldkjf':
    print(letter)

In [ ]:
for letter in 'abcd':
    print(f'value {letter}')
    


How about only the odd numbers? ...

In [ ]:
# hee hee good luck

Gotcha ! Haven't gone over math yet, have we.

## Math

In [ ]:
l = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for i in l: 
    print(i%2)

What the heck you think is happening here, huh?

In [ ]:
for i in range(10):
    print(i%2)

In [ ]:
l = [10, 100, 200]
for i in l:
    print(i//2)

What's goin on...

## If statements

In [ ]:
your_name =  '' #Enter your name

if your_name == 'Hao':
    print('浩白您好')
if your_name == 'Jungmin' or your_name == 'Ji':
    print('안녕하세요')
if your_name == 'Liam' or your_name == 'BossAssBitch':
    print('00110010010')

In [ ]:
# Write a program that tells me when my coffee is ready. 
# I don't know, use an if statement or something.

## Enumerate

In [ ]:
l = [1, 2, 3, 4, 5]
#print the index, value for l

## Grandmaster challenge

Uh oh guys, we've reached the **grandmaster** challenge, today's BEAST-MODE WRECKAGE moment. Best of luck. 

If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.

In [28]:
l = ['a', 'b', 'c']
l[2]

'c'

In [51]:
for i in range(len('abcd')): 
    print(i)

0
1
2
3


In [139]:
#List comprehenesion 



x = [i for i in range(1000) if i%2 == 0]
print(x)
print(sum(x))

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150, 152, 154, 156, 158, 160, 162, 164, 166, 168, 170, 172, 174, 176, 178, 180, 182, 184, 186, 188, 190, 192, 194, 196, 198, 200, 202, 204, 206, 208, 210, 212, 214, 216, 218, 220, 222, 224, 226, 228, 230, 232, 234, 236, 238, 240, 242, 244, 246, 248, 250, 252, 254, 256, 258, 260, 262, 264, 266, 268, 270, 272, 274, 276, 278, 280, 282, 284, 286, 288, 290, 292, 294, 296, 298, 300, 302, 304, 306, 308, 310, 312, 314, 316, 318, 320, 322, 324, 326, 328, 330, 332, 334, 336, 338, 340, 342, 344, 346, 348, 350, 352, 354, 356, 358, 360, 362, 364, 366, 368, 370, 372, 374, 376, 378, 380, 382, 384, 386, 388, 390, 392, 394, 396, 398, 400, 402, 404, 406, 408, 410, 412, 414, 416, 418, 420,

In [129]:
l = ['a', 'b', 'c', 'd']
#Only print every other element

for i in range(len(l)): 
    if i%2==0:
        print(l[i])

[print(l[i]) for i in range(len(l)) if i%2==0]
#Expected output: 
#a c

a
c
a
c


[None, None]

In [113]:
#Summing all even numbers below 1000
c=0
for i in range(1000):
    if i%2==0:
        c+=i
print(c)

249500


In [98]:
arr = [10, 4, 20, 3]

n = len(a)
for i in range(n): 
    for j in range(n-i-1): 
        if arr[j] > arr[j+1]: 
            arr[j], arr[j+1] = arr[j+1], arr[j]

print(arr)

[3, 4, 10, 20]


In [44]:
l = ['a', 'b', 'c']

for i, v in enumerate(l):
    print (i)
    

0
1
2


In [1]:
l = ['a', 'b', 'c', 'd']
list(enumerate(l))

[(0, 'a'), (1, 'b'), (2, 'c'), (3, 'd')]

In [ ]:
l = [ [0, 0, 'a'], [1, 1, 'b'], [2, 2, 'c'] ]
for i, j, k in l t:
    print(i, j, k)

In [ ]:
list = [ [0, 0], [1, 1], [2, 2] ]
i = 0 
for v in list:
    print (f'index{i} value{v}')
    i+=1

Let’s assume I have a list [“a”, “b”, “c”] and want to, for each element
of the list, print out its index-value pair.

In [ ]:
list = ['a','b','c','d']
i = 0
for v in list:
    print ('index ' + str(i) +' value ' + str(v))
    i +=1

In [ ]:
string = "abcd"
i = 0
for v in string:
    print('i is {} v is {}'+ str(i, v))
    print(f"index {i} value {v}")
    i += 1

Can I sum all the numbers below 100000 that are divisible by 414 or 4543?

Let’s assume I have a string “abc” and want to, for each element of the
string, print out its index-value pair.